core/modules/installer/installer.pages.inc

```diff
@@ -650,7 +650,12 @@ function installer_browser_installation_enable_form($form, &$form_state, $projec
   foreach ($projects as $project) {
     if ($project['type'] == 'module' && isset($modules[$project['name']])) {
       foreach ($modules as $module) {
-        if (isset($module->info['project']) && isset($modules[$project['name']]->info['project']) && $modules[$project['name']]->info['project'] == $module->info['project'] && $modules[$project['name']]->name != $module->name && !$module->info['hidden']) {
+        // Ignore stand-alone and hidden modules.
```

Comment:
We should also be ignoring any (sub)module that does not have a type defined in its .info file. So the next conditional should include a `|| !isset($module->info['type'])` ...while at it, can we make this even more understandable by doing something like:

```php
$is_hidden = !empty($module->info['hidden']);
$is_standalone = !isset($module->info['project']) || !isset($modules[$project['name']]->info['project']);
$no_project_type = !isset($module->info['type']);
...
```

Reply:
As explained in the issue comment, I'm not sure if core should catch violations like that (the documentation clearly says, "type" is required).

Reply:
I think it's feasible.